client/extensions/woocommerce/app/settings/payments/index.js

```diff
@@ -22,24 +32,79 @@ import SettingsPaymentsOnSite from './payments-on-site';
 class SettingsPayments extends Component {
 
 	static propTypes = {
+		isSaving: PropTypes.bool,
 		site: PropTypes.shape( {
 			slug: PropTypes.string,
 		} ),
 		className: PropTypes.string,
 	};
 
+	componentDidMount = () => {
+		const { site } = this.props;
+
+		if ( site && site.ID ) {
+			this.props.fetchSetupChoices( site.ID );
+		}
+	}
+
+	componentWillReceiveProps = ( newProps ) => {
+		const { site } = this.props;
+
+		const newSiteId = newProps.site ? newProps.site.ID : null;
+		const oldSiteId = site ? site.ID : null;
+
+		if ( oldSiteId !== newSiteId ) {
+			this.props.fetchSetupChoices( newSiteId );
+		}
+	}
+
+	onSave = () => {
+		const { translate, site, finishedInitialSetup } = this.props;
+
+		let successAction = successNotice(
+			translate( 'Payment settings saved.' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Payment Settings' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).